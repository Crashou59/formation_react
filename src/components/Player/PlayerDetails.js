import React from 'react';

export class PlayerDetails extends React.Component {

    render() {
        return <div className="Player-details panel">
            <p> Nom : {this.props.player.nom} </p>
            <p> Prenom : {this.props.player.prenom}</p>
            <p> Nickname : {this.props.player.nickname}</p>
        </div>;
    }
}