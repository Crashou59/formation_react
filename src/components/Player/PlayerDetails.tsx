import React from 'react';
import { PlayerType } from '../../type/PlayerType';


type MyProps = {
    player: PlayerType;
};
export class PlayerDetails extends React.Component<MyProps> {

    render() {
        return <div className="Player-details panel">
            <p> Nom : {this.props.player.nom} </p>
            <p> Prenom : {this.props.player.prenom}</p>
            <p> Nickname : {this.props.player.nickname}</p>
        </div>;
    }
}