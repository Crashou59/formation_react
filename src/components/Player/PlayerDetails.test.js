import React from 'react';
import { shallow } from 'enzyme';
import { PlayerDetails } from './PlayerDetails';

describe('PlayerDetails coponent', () => {

const firstPlayer = { nom: 'koala', prenom: 'koala2', nickname: 'nickname koala' };
const wrapper = shallow (<PlayerDetails player={firstPlayer} />);

it('Should render player details component', () => {
expect(wrapper.find('p').length).toEqual(3);
expect(wrapper.find(".Player-details").exists()).toEqual(true);
});

});