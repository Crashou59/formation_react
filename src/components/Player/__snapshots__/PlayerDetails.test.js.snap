// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDetails coponent renders correctly 1`] = `
<div
  className="Player-details"
>
  <p>
     Nom : 
    koala
     
  </p>
  <p>
     Prenom : 
    koala2
  </p>
  <p>
     Nickname : 
    nickname koala
  </p>
</div>
`;
